### PR TITLE
Avoid R package warning in `loadSupport()` if autoload is disabled

### PR DESCRIPTION
**Pull request by krlmlr**
in `loadSupport()`, if `_disable_autoload.R` exists.

### CODE REVIEW

**Comment by wch** [comment]
@krlmlr Can you provide instructions on how to reproduce the issue?

**Comment by krlmlr** [comment]
Create a Shiny app with a package structure (with `DESCRIPTION` and `NAMESPACE` files) that contains a file `R/_disable_autoload.R` and uses e.g. `pkgload::load_all()` to load the files in `R/` . The warning still occurs.

Do we need a test here?

**Comment by tanho63** [comment]
Minimal code for reproducing issue here:

```r
withr::with_dir(tempdir(check = TRUE),{
  dir.create("R", showWarnings = FALSE)
  # loadSupport documentation suggests adding a blank R/_disable_autoload.R file
  # will disable automatic loading of the R/ files
  file.create("R/_disable_autoload.R")
  # test that code was not loaded by adding a file
  writeLines(
    "cat('R/ code was sourced')",
    "R/code.R"
  )
  # loadSupport checks for a NAMESPACE file and/or a DESCRIPTION file to determine
  # if the directory contains a package, add NAMESPACE to trigger the check
  file.create("NAMESPACE")
  # add minimal app.R code
  writeLines(
    text = "
    shiny::shinyApp(
      ui = shiny::fluidPage(), 
      server = function(input, output, session) {}
    )
    ", 
    con = "app.R"
  )
  shiny::runApp()
  # still triggers warning, despite not actually running the code in question
})
```

File: R/shinyapp.R [hunk 1]
```diff
@@ -373,9 +376,6 @@ loadSupport <- function(appDir=NULL, renv=new.env(parent=globalenv()), globalren
   }
 
 
-  helpersDir <- file.path(appDir, "R")
-
-  disabled <- list.files(helpersDir, pattern="^_disable_autoload\\.r$", recursive=FALSE, ignore.case=TRUE)
   if (length(disabled) > 0){
     return(invisible(renv))
   }
```
Comment (tanho63) [inline R/shinyapp.R:381]: @krlmlr maybe simpler to move this early exit on disabled to L356 instead of extending the if clause?

Comment (gadenbuie) [reply R/shinyapp.R]: In the end, I think the warning was happening too early. By moving it down we can instead warn just before evaluating the code in the `R/` support directory.

File: R/shinyapp.R [hunk 2]
```diff
@@ -394,6 +385,26 @@ loadSupport <- function(appDir=NULL, renv=new.env(parent=globalenv()), globalren
...
+
+    has_desc_pkg <-
+      file.exists(descFile) &&
+      identical(as.character(read.dcf(descFile, fields = "Type")), "Package")
+  }
+
+  if (has_namespace || has_desc_pkg) {
+    warning(
+      "Loading R/ subdirectory for Shiny application, but this directory appears ",
+      "to contain an R package. Sourcing files in R/ may cause unexpected behavior."
```
Comment (krlmlr) [inline R/shinyapp.R:403]: Should this message point to a help file or something?

**Review by krlmlr — COMMENTED** [review]
Nice, thanks for picking this up!

File: R/shinyapp.R (hunk 2 above)
Comment (gadenbuie) [reply R/shinyapp.R:403]: Sure, something like this?

```suggestion
      "to contain an R package. Sourcing files in R/ may cause unexpected behavior. ",
      "See `?loadSupport` for more details."
```

**Review by cpsievert — APPROVED** [review]

**Pull request merged**